farm-staking/farm-staking/src/stake_farm.rs

```diff
@@ -71,22 +71,26 @@ pub trait StakeFarmModule:
         original_caller: ManagedAddress,
         payments: PaymentsVec<Self::Api>,
     ) -> EnterFarmResultType<Self::Api> {
-        let first_additional_payment_index = 1;
-        let boosted_rewards = match payments.try_get(first_additional_payment_index) {
-            Some(p) => self.claim_only_boosted_payment(&original_caller, &p),
-            None => EsdtTokenPayment::new(self.reward_token_id().get(), 0, BigUint::zero()),
+        let caller = self.blockchain().get_caller();
+
+        let boosted_rewards = self.claim_only_boosted_payment(&original_caller);
```

Comment:
adjust position and claim boosted. like everywhere. in farms as well. for entering and exiting. For merging we need only adjust_position.